src/test/java/com/iexec/worker/result/ResultServiceTests.java

```diff
@@ -156,14 +166,24 @@ void shouldGetWeb3ResultLink() {
     void shouldGetTeeWeb2ResultLinkSinceIpfs() {
         String storage = IPFS_RESULT_STORAGE_PROVIDER;
         String ipfsHash = "QmcipfsHash";
-
         when(iexecHubService.getTaskDescription(CHAIN_TASK_ID)).thenReturn(
-                TaskDescription.builder().chainTaskId(CHAIN_TASK_ID).isTeeTask(true).resultStorageProvider(storage).build());
-        when(resultProxyClient.getIpfsHashForTask(CHAIN_TASK_ID)).thenReturn(ipfsHash);
+                TaskDescription.builder()
+                        .chainTaskId(CHAIN_TASK_ID)
+                        .isTeeTask(true)
+                        .resultStorageProvider(storage)
+                        .resultStorageProxy(CUSTOM_RESULT_PROXY_URL)
+                        .build());
+        ResultProxyClient mockClient = mock(ResultProxyClient.class);
+        when(mockClient.getIpfsHashForTask(CHAIN_TASK_ID)).thenReturn(ipfsHash);
+        doReturn(mockClient)
+                .when(publicConfigurationService)
+                .createProxyClientFromURL(CUSTOM_RESULT_PROXY_URL);
```

Comment:
This syntax is not the main one and is only to be used in specific cases. We will prefer the following here:
```suggestion
        when(publicConfigurationService.createProxyClientFromURL(...)
            .thenReturn(resultProxyClient);
        when(resultProxy.getIpfsHashForTask(CHAIN_TASK_ID)).thenReturn(ipfsHash);
```

Reply:
See
https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#2 stubbing for generic case
https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#12 stubbing for specific case such as void methods